src/app/shared/components/validate-json-or-yaml/component.spec.ts

```diff
@@ -0,0 +1,230 @@
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeFalsy();
+    });
+
+    it('should render <mat-error> when data is truthy and isModelValid is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      component.isOnlyYAML = false;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element.textContent.trim()).toEqual('Please enter valid JSON or YAML');
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is true', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      component.isOnlyYAML = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element.textContent.trim()).toEqual('Please enter valid YAML');
+    });
+  });
+
+  describe('Component', () => {
+    it('should call _setIsDataValid() when ngOnChanges() is called', () => {
+      // @ts-expect-error
+      const spy = jest.spyOn(component, '_setIsDataValid');
+
+      component.ngOnChanges({
+        data: {firstChange: false, currentValue: '', previousValue: null, isFirstChange: () => false},
+      });
+
+      expect(spy).toHaveBeenCalled();
+    });
+
+    describe('_setIsDataValid() is called', () => {
+      it('should call _verifyYAML() when isOnlyYAML is true', () => {
+        component.isOnlyYAML = true;
+        const spy = jest.spyOn(component, <never>'_verifyYAML');
+
+        component['_setIsDataValid']();
+
+        expect(spy).toHaveBeenCalled();
+      });
+
+      it('should call _verifyYAML() when isOnlyYAML is false', () => {
+        component.isOnlyYAML = false;
+        const spy = jest.spyOn(component, <never>'_verifyYAML');
+
+        component['_setIsDataValid']();
+
+        expect(spy).toHaveBeenCalled();
+      });
+
+      it('should call _verifyJSON() when isOnlyYAML is false', () => {
+        component.isOnlyYAML = false;
+        const spy = jest.spyOn(component, <never>'_verifyJSON');
+
+        component['_setIsDataValid']();
+
+        expect(spy).toHaveBeenCalled();
+      });
+
+      it('should not call _verifyJSON() when isOnlyYAML is true', () => {
+        component.isOnlyYAML = true;
+        const spy = jest.spyOn(component, <never>'_verifyJSON');
+
+        component['_setIsDataValid']();
+
+        expect(spy).not.toHaveBeenCalled();
+      });
+
+      describe('set isModelValid when isOnlyYAML is false', () => {
```

Comment:
We don't have a `model` property anymore i guess it is `data`
Secondly, reword description -> description looks like for `it` block instead of complete spec

Reply:
property renamed to `isDataValid`,
rested `describe` block removed